ptr.py

```diff
@@ -104,10 +145,10 @@ def install_extra_dists(self, dist):
 		matching_extras = (
 			reqs
 			for (name, sep, marker), reqs in spec_extras
-			# never include extras that fail to pass marker eval
-			if self.marker_passes(marker)
 			# include unnamed extras or all if self.extras indicated
-			and (not name or self.extras)
+			if (not name or self.extras)
+			# never include extras that fail to pass marker eval
+			and (not marker or self.marker_passes(marker))
```

Comment:
I meant for `self.marker_passes` to encapsulate the full check. I think I want to fix the bug there. Was there a reason you switched the order of these tests?

Reply:
It just seemed more natural to me when I looked at fixing the conditions: first filter on the name, then check the condition.